benchmarks/mongo-es-speed/index.js

```diff
@@ -0,0 +1,152 @@
+        }
+    };
+
+    child.stdout.on('data', data => logHandler(data.toString('utf8')));
+    child.stderr.on('data', data => logHandler(data.toString('utf8')));
+
+    return done;
+}
+
+function dropCollection() {
+    return MongoClient.connect(config.MONGODB_CONNECTION_URL).then(db =>
+        new Promise(resolve => db
+            .collection(config.MONGODB_COLLECTION_NAME, { strict: true }, (err, collection) => (
+                err ? resolve() : collection.drop().then(resolve)
+        )))
+        .catch(err => log('Error while drop collection:', err))
+        .then(() => db.close())
+    );
+}
+
+function generateEvents(arr, i, storage) {
+    return dropCollection()
+        .then(() => prepareEvents(arr[i], totalEventProcessed))
+        .then(runLoadTests)
+        .then((data) => {
+            storage[arr[i]] = data;
+            if (i === arr.length - 1) {
+                return storage;
+            }
+            return generateEvents(arr, i + 1, storage);
+        });
+}
+
+function getDataString(arr, callback) {
+    return arr
+        .reduce((result, item) => `${result}${callback(item)},`, '')
+        .replace(/,$/, '\n');
+}
+
+function storeToCsv(filePath, points, data, dataGetter) {
+    return fs.writeFile(filePath,
+        getDataString(points, point => `Events count: ${point}`) +
+        getDataString(points, point => dataGetter(data, point))
+    );
+}
+
+function storeSummaryToXML(filePath, points, data) {
+    let dataString = '<?xml version="1.0" ?>\n<tabs>';
+
+    dataString += points.reduce((result, point) => `${result}<tab name="Events count: ${point}">`
+        + `<field name="Build time" value="${data[point].buildTime} ms" />`
+        + `<field name="Memory resident size" value="${Math.round(data[point].memory.rss / 1024 / 1024)} mb" />`
+        + `<field name="Memory heap total" value="${Math.round(data[point].memory.heapTotal / 1024 / 1024)} mb" />`
+        + `<field name="Memory heap used" value="${Math.round(data[point].memory.heapUsed / 1024 / 1024)} mb" />`
+        + '</tab>'
+    , '');
+
+    dataString += '</tabs>';
+
+    return fs.writeFile(filePath, dataString);
+}
+
+generateEvents(POINTS, 0, {})
+    .then(result => progress.tick(totalMaxValue) && result)
+    .then(result => Promise.all([
+        storeToCsv('./build-time.csv', POINTS, result, (data, point) => `${data[point].buildTime}`),
+        storeToCsv('./memory-rss.csv', POINTS, result, (data, point) =>
+            `${Math.round(data[point].memory.rss / 1024 / 1024)}`
+        ),
+        storeToCsv('./memory-heap-total.csv', POINTS, result, (data, point) =>
+            `${Math.round(data[point].memory.heapTotal / 1024 / 1024)}`
+        ),
+        storeToCsv('./memory-heap-used.csv', POINTS, result, (data, point) =>
+            `${Math.round(data[point].memory.heapUsed / 1024 / 1024)}`
+        ),
+        storeSummaryToXML('./summary.xml', POINTS, result)
+    ]))
+    .then(() => progressDonePromise)
```

Comment:
This construction is needed for UI in console benchmark launch, in which progress bar is present, and it should display complete state before end of process. If don't perform this, there will be illusion that benchmark crashed at about 96-98%, because progress-bar step is not such accurate